### PR TITLE
Hyperledger Fabric Operator for Kubernetes

### PR DESCRIPTION
**Pull request by dviejokfs**
Manage Hyperledger Fabric components at scale with a Kubernetes Operator.

I think there's a gap to deploy a Hyperledger Fabric network in Kubernetes, this project automates to automate the deployment.

The work will be focused on automating the creation of peers, orderers and CAs.

I have some ideas I would like to share such as developing a User Interface, and I think this could be integrated natively with the [fabric-operations-console](https://github.com/hyperledger-labs/fabric-operations-console) which was recently released.

### CODE REVIEW

File: labs/hlf-operator.md
Comment (arsulegai) [inline labs/hlf-operator.md:13]: [nitpick] The words such as `I think`, `this project`, `automates to automate` can be better framed. For example

```suggestion
There is a gap in the Hyperledger Fabric network deployment on Kubernetes. The proposal is to automate the deployment using the concept of Kubernetes Operators.
```

File: labs/hlf-operator.md
Comment (arsulegai) [inline labs/hlf-operator.md:17]: [nitpick] This statement can be rephrased as follows

```suggestion
The proposal is for the deployment of a Hyperledger Fabric network on Kubernetes. This proposal can be integrated natively with the [fabric-operations-console](https://github.com/hyperledger-labs/fabric-operations-console) which was recently released.
```

File: labs/hlf-operator.md
Comment (arsulegai) [inline labs/hlf-operator.md:25]: [Note on the sponsor section] The sponsor requirement is now an optional. A well connected member in the community (either TSC, SIG/WG Chair, Maintainer of a project) could be sponsor for the new Labs project. The idea was to introduce new members to the rest of the community and guide them through.

File: labs/hlf-operator.md
Comment (Dviejopomata) [reply labs/hlf-operator.md]: Added

File: labs/hlf-operator.md
Comment (Dviejopomata) [reply labs/hlf-operator.md]: Added

File: labs/hlf-operator.md
Comment (Dviejopomata) [reply labs/hlf-operator.md]: Added, though, It would be interesting to see if there's interest by HLF users, since probably the project needs a sponsor, do you know how to approach this?

Comment (arsulegai) [reply labs/hlf-operator.md]: Sure, easiest way you can reach out to maintainers of a project is to find their mail ids in the `MAINTAINERS.md` file of the project repository. I am sure Labs stewards here will be happy to help. I am happy to guide you through as well. My RocketChat goes by the same name @arsulegai .
There was another proposal in the past which is similar to what I see here. Maybe this proposal has potential to grow community, integrate with other projects.

**Review by tkuhrt — APPROVED** [review]
Approving this lab. Although we will need to ensure that the DCO signoff on the existing repo is handled prior to transferring the code to labs.

**Review by arsulegai — APPROVED** [review]

**Review by lehors — COMMENTED** [review]
LGTM. I'm glad the connection with the fabric-operations-console was highlighted. You should also look at Cello which includes such an operator and could probably be leveraged to get started.

**Comment by dviejokfs** [comment]
> LGTM. I'm glad the connection with the fabric-operations-console was highlighted. You should also look at Cello which includes such an operator and could probably be leveraged to get started.

Great! 

Could we set up a meeting or a chat somewhere to have a review with Cello and other projects?

**Comment by dviejokfs** [comment]
@lehors 
I see that the fabric operator it's outdated since the last modification was 12 months ago. It doesn't support Hyperledger Fabric 2.3 with the new channel participation API for the ordering service.

https://github.com/hyperledger/cello/tree/main/src/agent/fabric-operator

**Comment by lehors** [comment]
You're right, the operator which is part of Cello is outdated so maybe there is nothing you can leverage from it but I wanted you to know it's out there. If your implementation is more advanced it would be good to let the Cello people know so maybe they can just leverage yours instead of maintaining one on their own.

**Review by lehors — APPROVED** [review]
LGTM

**Comment by dviejokfs** [comment]
Sorry to comment on this old PR.
@lehors  @ryjones @arsulegai How can we move this forward?

**Comment by tkuhrt** [comment]
> Sorry to comment on this old PR. @lehors @ryjones @arsulegai How can we move this forward?

@dviejokfs : You will need to take care of the DCO on your existing repository. As I noted earlier:

> Hyperledger requires that all contributed code has a DCO sign off. Looking at these repositories, it appears that this is not the case. There are two options that we normally consider for fixing this:
> 
> 1. Amend existing commits in the repository to include DCO sign off
> 2. Manually transfer the code from the existing repo to the newly created repo within the hyperledger-labs GitHub organization.

Which step do you plan to take?

**Comment by dviejokfs** [comment]
Thank you, @tkuhrt.

I have a couple of questions:

Will the issues and stars be migrated if they are transferred to a new organization?

Also, since this repository is in an already existing organization, does the process change?

Thank you

**Comment by tkuhrt** [comment]
Hi, @dviejokfs : Please see this page for [information about repository transfers](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository#whats-transferred-with-a-repository). Please transfer the repository to @ryjones, and he can move it to the hyperledger-labs organization.

**Comment by dviejokfs** [comment]
Ok @tkuhrt, I'm planning to transfer it this Saturday, let me know how can we coordinate to make this process as smooth as possible.

**Comment by arsulegai** [comment]
@dviejokfs you could coordinate with @ryjones on discord in #community-architects channel. You can join the discord at https://discord.gg/hyperledger

**Pull request merged**
